### ISSUE TITLE
Change Tor identity (IP address)

### ISSUE THREAD

**Issue by rightaway** [Mowje/node-ths#12]
It would be great if node-ths included an API method to change the Tor identity (IP address), like what is in this gist https://gist.github.com/estliberitas/4448496 (linked from http://stackoverflow.com/questions/19498796/how-to-let-tor-change-ip-in-nodejs), so you could do something like ths.changeIdentity().

**Comment by mattcollier** [comment]
Hi rightaway,

Just to clarify.  The NEWYM control port command tells your tor client to rebuild new circuits.  The command is documented here: https://gitweb.torproject.org/torspec.git/tree/control-spec.txt#n414

When you are referring to 'tor identity', I believe you mean the IP address for your exit node, is this correct?  If so, yes you would very likely have a different exit node (with its corresponding IP) after rebuilding circuits.  Circuits are rebuilt automatically every 10 minutes by default.

I'm not sure how the NEWYM command relates to Tor hidden services.  It seems to me that it would only affect outgoing connections. 

You could use the code from the gist you referenced to communicate with the tor instance that node-ths has started for you.  I feel that the cookie authentication being used by your gist is overly complicated, and node-ths sets up the control port to use a password.

You can use ths.controlPass() to determine the current password.  The default control port number is 9998.

Also, I have created this module as a companion to node-ths for sending arbitrary control port commands: https://github.com/mattcollier/tor-control

**Comment by rightaway** [comment]
Yes I believe the NEWYM command relates to outgoing connections and doesn't have much to do with hidden services. Thanks for creating this!
